LoopBack/LBModel.m

```diff
@@ -96,18 +96,38 @@ - (LBModel *)modelWithDictionary:(NSDictionary *)dictionary {
 
     [[model _overflow] addEntriesFromDictionary:dictionary];
 
-    [dictionary enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+    for (NSString *key in dictionary) {
+        id obj = dictionary[key];
         SEL setter = NSSelectorForSetter(key);
 
+        // if obj is a NSNumber (including a boolean value) try to use a setter for the primitive type
+        if ([obj isKindOfClass:[NSNumber class]]) {
```

Comment:
Is there any unit-test covering this new addition? If there isn't, could you please add one?